src/jsCore.ts

```diff
@@ -0,0 +1,497 @@
+  }
+}
+
+function createTracker(
+  configuration: InitTrackerConfiguration,
+  emitCallback?: (e: Payload) => void
+): Promise<void> {
+  // create an emit callback if not given
+  const emitter = emitCallback ?? createEmitCallback(configuration.networkConfig);
+
+  // the tracker core does not provide an option to set the duid, so we need to add custom
+  let domainUserId: string | undefined;
+  const setDomainUserId = (userId: string | undefined): void => {
+    domainUserId = userId;
+  };
+  let networkUserId: string | undefined;
+  const setNetworkUserId = (userId: string | undefined): void => {
+    networkUserId = userId;
+  };
+
+  // initialize the tracker core
+  const core = trackerCore({
+    base64: configuration.trackerConfig?.base64Encoding ?? true,
+    callback: (payload: PayloadBuilder) => {
+      if (domainUserId != null) {
+        payload.add('duid', domainUserId);
+      }
+      if (networkUserId != null) {
+        payload.add('tnuid', networkUserId);
+      }
+      const builtPayload = payload.build();
+      emitter(builtPayload);
+    },
+  });
+  const tracker = { ...core, setDomainUserId, setNetworkUserId };
+  trackers[configuration.namespace] = tracker;
+
+  // update tracker properties to reflect subject and tracker info
+  updateTrackerProperties(tracker, configuration);
+
+  return <Promise<void>>Promise.resolve();
+}
+
+function removeTracker(details: { tracker: string }): Promise<boolean> {
+  delete trackers[details.tracker];
+  return <Promise<boolean>>Promise.resolve(true);
+}
+
+function removeAllTrackers(): Promise<boolean> {
+  trackers = {};
+  return <Promise<boolean>>Promise.resolve(true);
+}
+
+function forTracker(
+  namespace: string | null,
+  callback: (tracker: Tracker) => void
+): void {
+  const tracker = namespace != null ? trackers[namespace] : Object.values(trackers)[0];
+
+  if (tracker) {
+    callback(tracker);
+  } else {
+    errorHandler(new Error('No such tracker found.'));
+  }
+}
+
+function trackSelfDescribingEvent(details: {
+  tracker: string | null;
+  eventData: SelfDescribing;
+  contexts: EventContext[];
+}): Promise<void> {
+  forTracker(details.tracker, (tracker) => {
+    tracker.track(
+      buildSelfDescribingEvent({
+        event: details.eventData,
+      }),
+      details.contexts
+    );
+  });
+  return <Promise<void>>Promise.resolve();
+}
+
+function trackStructuredEvent(details: {
+  tracker: string | null;
+  eventData: StructuredProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  forTracker(details.tracker, (tracker) => {
+    tracker.track(buildStructEvent(details.eventData), details.contexts);
+  });
+  return <Promise<void>>Promise.resolve();
+}
+
+function trackScreenViewEvent(details: {
+  tracker: string | null;
+  eventData: ScreenViewProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  forTracker(details.tracker, (tracker) => {
+    tracker.track(buildScreenView(details.eventData), details.contexts);
```

Comment:
Just noticed that [the screen view schema used in native](https://github.com/snowplow/snowplow-objc-tracker/blob/ec912a566073615ebcc9dc3a4c6b96a1cc6ed7e7/Snowplow/Internal/SPTrackerConstants.m#L57) is different than the one used by [`buildScreenView`](https://github.com/snowplow/snowplow-javascript-tracker/blob/4dec01b9a314ca11d88372afa4b2dc67a431ee0d/libraries/tracker-core/src/core.ts#L814) in JavaScript tracker. I did not know that! This means we'd need to switch to `trackSelfDescribingEvent`?

Reply:
Life-saving catch again, thanks Ada! I completely missed that the events are tracked with the other screen view schema. Fixed it.